terraform/modules/tf_stream_alert_cloudwatch/iam.tf

```diff
@@ -0,0 +1,87 @@
+# IAM Role: Allows CloudWatch Logs to put data into
+# this cluster's default Kinesis stream
+resource "aws_iam_role" "cloudwatch_subscription_role" {
+  name = "stream_alert_${var.cluster}_cloudwatch_subscription_role"
+
+  assume_role_policy = "${data.aws_iam_policy_document.cloudwatch_logs_assume_role_policy.json}"
+}
+
+// IAM Policy Document: AssumeRole for CloudWatch Logs
+data "aws_iam_policy_document" "cloudwatch_logs_assume_role_policy" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["logs.${var.region}.amazonaws.com"]
+    }
+  }
+}
+
+// IAM Policy: Write to Kinesis
+resource "aws_iam_role_policy" "cloudwatch_kinesis_wo" {
+  name = "stream_alert_${var.cluster}_write_cloudwatch_to_kinesis"
+  role = "${aws_iam_role.cloudwatch_subscription_role.id}"
+
+  policy = "${data.aws_iam_policy_document.cloudwatch_put_kinesis_events.json}"
+}
+
+// IAM Policy Document: Write to Kinesis
+data "aws_iam_policy_document" "cloudwatch_put_kinesis_events" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "kinesis:PutRecord",
```

Comment:
Does it need `PutRecords` as well?

Reply:
The docs only show needing `PutRecord`. See step # 5 here: https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/CreateDestination.html

I also tested this with the perms they describe and it works.

Reply:
Interesting that it does no batching 🤔